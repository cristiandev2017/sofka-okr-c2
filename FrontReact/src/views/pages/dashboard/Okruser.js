import React, {Fragment} from "react";
//Circular Progressbar
import { CircularProgressbar } from "react-circular-progressbar";
import "react-circular-progressbar/dist/styles.css";

function Okruser({ okr }) {
  //Valido que vengan los datos
  let title = "";
  let objetivo = "";
  let description = "";
  let areainCharge = "";
  let progress = 0;
  if (okr !== null) {
    progress = okr[0].progress;
    title = okr[0].title;
    description = okr[0].description;
    areainCharge = okr[0].areainCharge;
    objetivo = okr[0].objetivo;
  }

  return (
    <Fragment>
      {okr ? (
        <div>
          <div style={{ width: "16%" }}>
            <CircularProgressbar value={progress} text={`${progress}%`} />
          </div>
          <div>
            <table className="table">
              <thead>
                <tr>
                  <th scope="col">#</th>
                  <th scope="col">Titulo</th>
                  <th scope="col">Descripcion</th>
                  <th scope="col">Area encargada</th>
                  <th scope="col">objetivo</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <th scope="row">1</th>
                  <td>{title}</td>
                  <td>{description}</td>
                  <td>{areainCharge}</td>
                  <td>{objetivo}</td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      ) : (
        "Seleccione un OKR"
      )}
    </Fragment>
  );
}

export default Okruser;
