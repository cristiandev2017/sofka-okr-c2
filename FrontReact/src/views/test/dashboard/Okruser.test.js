import React from 'react';
import renderer from 'react-test-renderer';
import Okruser from './../../pages/dashboard/Okruser';


it('renders correctly', () => {
  const okr = {
    id:"1",
    title:"title of okr",
    description:"desds",
    areainCharge:"Agile services",
    krs:[
    
    ]
   }

  const tree = renderer
    .create(<Okruser okr={okr}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});