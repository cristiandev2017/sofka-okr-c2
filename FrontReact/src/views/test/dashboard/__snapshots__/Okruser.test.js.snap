// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Titulo
          </th>
          <th
            scope="col"
          >
            Descripcion
          </th>
          <th
            scope="col"
          >
            Area encargada
          </th>
          <th
            scope="col"
          >
            objetivo
          </th>
          <th
            scope="col"
          >
            Progreso
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            title of okr
          </td>
          <td>
            desds
          </td>
          <td />
          <td />
          <td>
            <div
              style={
                Object {
                  "width": "60%",
                }
              }
            >
              <svg
                className="CircularProgressbar "
                data-test-id="CircularProgressbar"
                style={Object {}}
                viewBox="0 0 100 100"
              >
                <path
                  className="CircularProgressbar-trail"
                  d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
                  fillOpacity={0}
                  strokeWidth={8}
                  style={
                    Object {
                      "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                      "strokeDashoffset": "0px",
                    }
                  }
                />
                <path
                  className="CircularProgressbar-path"
                  d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
                  fillOpacity={0}
                  strokeWidth={8}
                  style={
                    Object {
                      "strokeDasharray": "289.02652413026095px 289.02652413026095px",
                      "strokeDashoffset": "289.02652413026095px",
                    }
                  }
                />
                <text
                  className="CircularProgressbar-text"
                  style={Object {}}
                  x={50}
                  y={50}
                >
                  undefined%
                </text>
              </svg>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            scope="col"
          >
            Titulo
          </th>
          <th
            scope="col"
          >
            Descripcion KR
          </th>
          <th
            scope="col"
          >
            Fecha de Inicio
          </th>
          <th
            scope="col"
          >
            Fecha Final
          </th>
          <th
            scope="col"
          >
            Porcentaje OKR
          </th>
          <th
            scope="col"
          >
            Progreso
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
